src/create-dom-event.ts

```diff
@@ -0,0 +1,104 @@
+import eventTypes from 'dom-event-types'
+
+const keyCodesByKeyName = {
+  backspace: 8,
+  tab: 9,
+  enter: 13,
+  esc: 27,
+  space: 32,
+  pageup: 33,
+  pagedown: 34,
+  end: 35,
+  home: 36,
+  left: 37,
+  up: 38,
+  right: 39,
+  down: 40,
+  insert: 45,
+  delete: 46
+}
+
+function getEventProperties(eventParams) {
+  const { modifier, meta, options } = eventParams
+  const keyCode =
+    keyCodesByKeyName[modifier] ||
+    options.key ||
+    options.keyCode ||
+    options.code
+
+  return {
+    ...options, // What the user passed in as the second argument to #trigger
+    bubbles: meta.bubbles,
+    meta: meta.cancelable,
+    // Any derived options should go here
+    key: keyCode,
```

Comment:
`key`, `keyCode`, `code` are 3 different things with different values, they shouldn't get the same value

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/KeyboardEvent

Reply:
I have a couple of questions about this, surely someone with more knowledge of events than me can solve them.

If I am not mistaken, currently in 'vue-test-utils' this distinction is not being made. With the current implementation it seems that events are being fired without problem (more tests should be added to confirm that).

It is clear that the more real the function, the fewer unexpected errors, but do the three values always have to be defined? Right now one of them is correct and the rest are usually `undefined`.

Assuming that each key has the correct value, should the expected results of these values be as follows?

`trigger('keydown', { key: 'Enter' })` and `trigger('keydown.enter')`
- `key`: 'Enter', `code`: 'Enter', `keyCode`: 13

`trigger('keydown', { keyCode: '65' })`
- `key`: 'A', `code`: 'KeyA', `keyCode`: 65

Reply:
I don't know very much about the intricacies of `KeyboardEvent`.  I'd say write the tests how you would want to write them, and implement whatever we need to get them to pass. Ideally we should match whatever Vue lets you do. Eg, you can do `keyup.enter` in Vue, so we should support `trigger('keyup.enter')`.

I think this was something VTU beta did fairly well, if we can just replicate that (which it looks like you are doing) I think we will be in a good place.

We can definitely make improvements after that, I'd say for now we aim for basic feature parity with beta (we are pretty close) then we can dig deeper if there are any outstanding problems.

Reply:
> We can definitely make improvements after that, I'd say for now we aim for basic feature parity with beta (we are pretty close) then we can dig deeper if there are any outstanding problems.

Perfect, I'll focus on a first version that works properly and then we'll think about what improvements can be made